packages/expo-gl-cpp/cpp/EXGLContext.cpp

```diff
@@ -1,21 +1,26 @@
 #include "EXGLContext.h"
+#include "EXPlatformUtils.h"
 
 namespace expo {
 namespace gl_cpp {
 
 constexpr const char *OnJSRuntimeDestroyPropertyName = "__EXGLOnJsRuntimeDestroy";
 
 EXGLContext::EXGLContext(jsi::Runtime &runtime, UEXGLContextId ctxId) : ctxId(ctxId) {
-  auto viewport = prepareOpenGLESContext();
-  createWebGLRenderer(runtime, this, viewport);
-  tryRegisterOnJSRuntimeDestroy(runtime);
-
-  jsi::Value workletRuntimeValue = runtime.global().getProperty(runtime, "_WORKLET_RUNTIME");
-  if (workletRuntimeValue.isNumber()) {
-    jsi::Runtime &workletRuntime =
-        *reinterpret_cast<jsi::Runtime *>(static_cast<uintptr_t>(workletRuntimeValue.getNumber()));
-    createWebGLRenderer(workletRuntime, this, viewport);
-    tryRegisterOnJSRuntimeDestroy(workletRuntime);
+  try {
+    auto viewport = prepareOpenGLESContext();
+    createWebGLRenderer(runtime, this, viewport);
+    tryRegisterOnJSRuntimeDestroy(runtime);
+
+    jsi::Value workletRuntimeValue = runtime.global().getProperty(runtime, "_WORKLET_RUNTIME");
+    if (workletRuntimeValue.isNumber()) {
+      jsi::Runtime &workletRuntime = *reinterpret_cast<jsi::Runtime *>(
+          static_cast<uintptr_t>(workletRuntimeValue.getNumber()));
+      createWebGLRenderer(workletRuntime, this, viewport);
+      tryRegisterOnJSRuntimeDestroy(workletRuntime);
+    }
+  } catch (const std::runtime_error &err) {
+    EXGLSysLog("%s", err.what());
```

Comment:
it depends at what time it would fail, but in most cases, gl object creation would fail or created object would not have appropriate methods. 

I don't expect any error here to happen, (maybe if runtime is heavily modified, or maybe old jsc version), but without that catch, the app is just crashing, normally c++ error would be converted to js error and passed to js, but his code is called from java/objc.

Reply:
> gl object creation would fail or created object would not have appropriate methods

in such case and plus the error log, that would be enough to troubleshooting.
thanks!